format/proto/types.proto

```diff
@@ -1 +1,36 @@
-option java_package = "org.corfudb.format";
\ No newline at end of file
+option java_package = "org.corfudb.format";
+
+enum DataType {
+    DATA = 0;
+    EMPTY = 1;
+    HOLE = 2;
+    TRIMMED = 3;
+}
+
+message LogEntry {
+    optional DataType data_type = 1;
+    optional bytes data = 2;
+    optional int64 global_address = 3;
+    optional int64 rank = 4;
+    optional bool commit = 5;
+    repeated string	streams = 6;
+    map<string, int64> logical_addresses = 7;
+    map<string, int64> backpointers = 8;
+}
+
+message LogHeader {
+    optional int32 version = 1;
+    optional bool verify_checksum = 2;
+}
+
+message LogRecord {
+    // The checksum is computed over fields 2-4
+    optional int64 address = 1;
+    optional int32 length = 2;
+    optional LogEntry entry = 3;
```

Comment:
Since LogEntry is encapsulated, then if there is an address mismatch we cannot skip the entry because it needs to be parsed as part of the LogRecord message. Maybe Make LogRecord a MetaData message that doesnt encapsulate the LogEntry, so the format would looke like  FileHeader|MetaData|LogEntry|. That why only MetaData needs to be parsed and the LogEntry can be skipped if theres an address mismatch.